### DISCUSSION TITLE
How to use lastest grammar queryRaw wildcard % in mysql?

### DISCUSSION THREAD

**Discussion by jun-ho-Kim** [prisma/prisma#11616] [Q&A] [answered]
```
const orderTest = await this.prisma.$queryRaw(`
				SELECT IFNULL(SUM(total_amount), 0) total
				FROM t_order_history T_ORDER_HISTORY
				INNER JOIN t_store T_STORE
				ON T_ORDER_HISTORY.store_id = T_STORE.store_id
				WHERE T_STORE.op_id='${args.op_id}'
				AND T_STORE.fc_id='${args.fc_id}'
				AND order_time like '${dateFormat(yesterday, 'yyyy-mm-dd')}%'
				AND order_state='completed'
		`)
```
 I want to use lastest grammar queryRaw wildcard % in mysql
So, I revised it to primsa lastest grammar code below

```
			const parseOrderTest = await this.prisma.$queryRaw`
				SELECT IFNULL(COUNT(total_amount), 0) count
				FROM t_order_history T_ORDER_HISTORY
				INNER JOIN t_store T_STORE
				ON T_ORDER_HISTORY.store_id = T_STORE.store_id
				WHERE T_STORE.op_id=${args.op_id}
				AND T_STORE.fc_id=${args.fc_id}
				AND order_time like ${dateFormat(yesterday, 'yyyy-mm-dd')}%
				AND order_state='completed'
		`
```

However, the following error occurred.

` "message": "\nInvalid `prisma.queryRaw()` invocation:\n\n\n  Raw query failed. Code: `N/A`. Message: `N/A`",`

Please, How to use queryRaw`SELECT ... FROM ... WHERE ... % ...`

**Comment by pantharshit00** [comment, marked as answer]
The older code that you were using was prone to SQL injection. The new API method is trying to prevent SQL injection by replacing every template string variable with a prepared statement variable. So in order to format your arguments properly, you will need to do the following. 
```ts
const parseOrderTest = await this.prisma.$queryRaw`
				SELECT IFNULL(COUNT(total_amount), 0) count
				FROM t_order_history T_ORDER_HISTORY
				INNER JOIN t_store T_STORE
				ON T_ORDER_HISTORY.store_id = T_STORE.store_id
				WHERE T_STORE.op_id=${args.op_id}
				AND T_STORE.fc_id=${args.fc_id}
				AND order_time like ${`${dateFormat(yesterday, 'yyyy-mm-dd')}%`}
				AND order_state='completed'
`;
```
Notice we omited all the quotes as those will be replaced by prepared statement variables. 

It will produce the following query and send the parameters:
```sql
SELECT IFNULL(COUNT(total_amount), 0) count
				FROM t_order_history T_ORDER_HISTORY
				INNER JOIN t_store T_STORE
				ON T_ORDER_HISTORY.store_id = T_STORE.store_id
				WHERE T_STORE.op_id=$1
				AND T_STORE.fc_id=$2
				AND order_time like $3
				AND order_state='completed'
```

Let me know if this solves the issue.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
